t/response.t

```diff
@@ -236,3 +236,64 @@ qr/\[TRACE\s+\d+ content_by_lua\(nginx\.conf:\d+\):5 (?:loop|-> \d+)\]/
 [error]
  -- NYI:
 stitch
+
+
+
+=== TEST 8: don't generate Content-Type header when setting other response headers
+--- http_config eval: $::HttpConfig
+--- config
+    location = /t {
+        default_type text/html;
+        rewrite_by_lua_block {
+            ngx.header.blah = "foo"
+        }
+        proxy_pass http://127.0.0.1:$TEST_NGINX_SERVER_PORT/backend;
+    }
+
+    location = /backend {
+        content_by_lua_block {
+            ngx.say("foo")
+        }
+        header_filter_by_lua_block {
+            ngx.header.content_type = nil
+        }
+    }
+--- request
+GET /t
+--- response_body
+foo
+--- response_headers
+blah: foo
+Content-Type:
+--- no_error_log
+[error]
+
+
+
+=== TEST 9: don't generate Content-Type header when getting other response headers
+--- http_config eval: $::HttpConfig
+--- config
+    location = /t {
+        default_type text/html;
+        rewrite_by_lua_block {
+            local h = ngx.header.content_length
+        }
+        proxy_pass http://127.0.0.1:$TEST_NGINX_SERVER_PORT/backend;
+    }
+
+    location = /backend {
+        content_by_lua_block {
+            ngx.say("foo")
+        }
+        header_filter_by_lua_block {
+            ngx.header.content_type = nil
+        }
+    }
+--- request
+GET /t
+--- response_body
+foo
+--- response_headers
+Content-Type:
+--- no_error_log
+[error]
```

Comment:
imho it is fine keeping these tests here, but they are duplicate of the sister PR's tests. Running the ngx_lua test suite with the resty.core module is also a way to run these tests.

In fact, changes like [this one](https://github.com/openresty/lua-nginx-module/pull/1422/files#diff-8593203ffbe2a651fb1761cebff6c139R1409) are only tested from within the ngx_lua test suite (when running with the resty.core module). So if the goal is to port ngx.resp tests to this module (resty.core), then that is a missing test. Is that the goal?

Reply:
@thibaultcha 
I decide to remove the duplicate tests, as we always have to run

```shell
TEST_NGINX_INIT_BY_LUA="package.path = '$PWD/../lua-resty-core/lib/?.lua;$PWD/../../git/lua-resty-lrucache/lib/?.lua;' .. (package.path or '') require('resty.core')" prove ...
```

to ensure all the test cases are passed.

Reply:
Yes indeed :) Great choice, thank you!